src/components/trading_order.cpp

```diff
@@ -0,0 +1,97 @@
+#include "trading_order.h"
+
+TradingOrder::TradingOrder() {
+  trading_order_time_stamp = -1;
+}
+
+void TradingOrder::setSymbol(QString symbol) {
+  this->symbol = symbol;
+}
+
+void TradingOrder::setQuantity(qreal quantity) {
+  this->quantity = quantity;
+}
+
+void TradingOrder::setStrategy(TradingStrategy strategy) {
+  this->strategy = strategy;
+}
+
+void TradingOrder::setOrderType(TradingOrderType order_type) {
+  this->order_type = order_type;
+}
+
+void TradingOrder::setAction(TradingAction action) {
+  this->action = action;
+}
+
+void TradingOrder::setTiming(TradingTiming timing) {
+  this->timing = timing;
+}
+
+void TradingOrder::setLimitPrice(qreal limit_price) {
+  this->limit_price = limit_price;
+}
+
+void TradingOrder::setTradingTime(qint64 trading_order_time_stamp) {
+  this->trading_order_time_stamp = trading_order_time_stamp;
+}
+
+void TradingOrder::setValuePerQuantity(qreal value_per_quantity) {
+  this->value_per_quantity = value_per_quantity;
+}
+
+double TradingOrder::valuation() {
+  return value_per_quantity * quantity;
+}
+
+void TradingOrder::read(const QJsonObject &json) {
+  if (json.contains("symbol") && json["symbol"].isString()) {
+    symbol = json["symbol"].toString();
+  }
+
+  if (json.contains("quantity") && json["quantity"].isDouble()) {
+    quantity = json["quantity"].toInt();
+  }
+
+  if (json.contains("strategy") && json["strategy"].isString()) {
+    symbol = json["strategy"].toString();
+  }
+
+  if (json.contains("order_type") && json["order_type"].isString()) {
+    symbol = json["order_type"].toString();
```

Comment:
Need to convert order_type back into enum and assign it to order_type 

I already wrote the if else blocks for this in my PR so feel free to copy paste.

Reply:
This is a bug in the PR since I assign everything to symbol.